.travis.yml

```diff
@@ -12,58 +23,89 @@ cache:
     else
-      echo "=== TESTING PHING ==="
-      cd test
-      ../bin/phing -Dtests.codecoverage=true -listener "phing.listener.StatisticsListener"
-      cd ..
-    fi
-after_success:
-  - |
-    if [[ "$PHPCS" != "TRUE" ]]; then
-      if [[ "$TRAVIS_BRANCH" == "master" ]]; then
-        bash <(curl -s https://codecov.io/bash)
-      fi
-      wget https://scrutinizer-ci.com/ocular.phar
-      php ocular.phar code-coverage:upload --format=clover ./test/reports/clover-coverage.xml
+      composer install --optimize-autoloader --prefer-dist --no-progress --no-interaction --no-suggest;
     fi
 
-php:
-  - 7.1
-  - 7.2
-  - 7.3
-  - 7.4snapshot
-  - master
+script:
+  - echo "=== TESTING PHING ==="
+  - cd test
+  - ../bin/phing -debug -Dtests.codecoverage=false -listener "phing.listener.StatisticsListener"
+  - cd ..
 
-matrix:
-  include:
-    - php: 7.2
-      env: PHPCS="TRUE"
-    - php: 7.1
-      env: DEPS="LOW"
-  fast_finish: true
+jobs:
   allow_failures:
     - php: 7.4snapshot
     - php: master
+    - php: nightly
+  include:
+    - php: nightly
+      env: DEPS="IGNORE"
+      install: composer update --optimize-autoloader --prefer-dist --prefer-stable --no-progress --no-interaction --no-suggest --ignore-platform-reqs
+
+    - stage: test with coverage
+      php: 7.2
+      env: DEPS="NORMAL"
+      before_install:
+        - phpenv config-add .travis.php.ini
+        - phpenv rehash
+        - echo 'opcache.enable=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+        - echo 'opcache.enable_cli=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+        - echo 'zend.assertions=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+        - echo 'assert.exception=On' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+        - set -e
+        - pear config-set php_dir $(php -r 'echo substr(get_include_path(),2);')
+        - echo "=== SETTING GIT IDENTITY ==="
+        - git config --global user.email "travis-ci-build@phing.info"
+        - git config --global user.name "Phing Travis Builder"
+        - composer self-update
+      script:
+        - echo "=== TESTING PHING ==="
+        - cd test
+        - ../bin/phing -Dtests.codecoverage=true -listener "phing.listener.StatisticsListener"
+        - cd ..
+      after_success:
+        - bash <(curl -s https://codecov.io/bash) -f ./test/reports/clover-coverage.xml
+        - wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=clover ./test/reports/clover-coverage.xml
+
+    - stage: composer validate
+      os: linux
+      php: 7.2
+      env: DEPS="NORMAL"
+      script:
+        - composer validate
+
+    - stage: static code analysis
+      os: linux
+      php: 7.2
+      env: DEPS="NORMAL"
+      script: bin/phpcs -s --standard=./ruleset.xml classes
```

Comment:
Please add [`--config-set ignore_warnings_on_exit 1`](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#ignoring-warnings-when-generating-the-exit-code) to avoid the build to fail on warnings.

Reply:
I would prefer the `-n` option as it was before.